Packaging/Debian/nextspace-applications-0.85+20200520.gitdcb7a3e6/debian/patches/add-wm-resources.patch

```diff
@@ -0,0 +1,437 @@
+Description: <short summary of the patch>
+ TODO: Put a short summary on the line above and replace this paragraph
+ with a longer explanation of this change. Complete the meta-information
+ with other relevant fields (see below for details). To make it easier, the
+ information below has been extracted from the changelog. Adjust it or drop
+ it.
+ .
+ nextspace-applications (0.85+git7f12a6e-1) unstable; urgency=medium
+ .
+   * Initial release
+Author: Patrick Georgi <patrick@georgi.software>
+
+---
+The information above should follow the Patch Tagging Guidelines, please
+checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
+are templates for supplementary fields that you might want to add:
+
+Origin: <vendor|upstream|other>, <url of original patch>
+Bug: <url in upstream bugtracker>
+Bug-Debian: https://bugs.debian.org/<bugnumber>
+Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
+Forwarded: <no|not-needed|url proving that it has been forwarded>
+Reviewed-By: <name and email of someone who approved the patch>
+Last-Update: 2020-05-21
+
+--- nextspace-applications-0.85+git7f12a6e.orig/Workspace/GNUmakefile.postamble
++++ nextspace-applications-0.85+git7f12a6e/Workspace/GNUmakefile.postamble
+@@ -58,10 +58,10 @@ after-install::
+ 	@ echo " Copying WM resources into $(WM_RESOURCES)..."
+ 	@ mkdir -p $(WM_RESOURCES)
+ 	@ cp $(WM_DIR)/WINGs/Resources/Images.tiff $(WM_RESOURCES)
+-	@ cp -f /etc/skel/Library/Preferences/.WindowMaker/WMGLOBAL $(WM_RESOURCES)
+-	@ cp -f /etc/skel/Library/Preferences/.WindowMaker/WMState $(WM_RESOURCES)
+-	@ cp -f /etc/skel/Library/Preferences/.WindowMaker/WMWindowAttributes $(WM_RESOURCES)
+-	@ cp -f /etc/skel/Library/Preferences/.WindowMaker/WindowMaker $(WM_RESOURCES)
++	@ cp -f $(WM_DIR)/WINGs/Resources/WMGLOBAL $(WM_RESOURCES)
++	@ cp -f $(WM_DIR)/WINGs/Resources/WMState $(WM_RESOURCES)
++	@ cp -f $(WM_DIR)/WINGs/Resources/WMWindowAttributes $(WM_RESOURCES)
++	@ cp -f $(WM_DIR)/WINGs/Resources/WindowMaker $(WM_RESOURCES)
```

Comment:
It may lead to misconfiguration in the future. My idea was to maintain default configuration files for user hoome directory in System/etc/skel regrdless what options is set inside Workspace sources. I may remove WINGs/Resources completely from Workspace (as well as WINGs itself).

Reply:
The configuration mechanism needs a larger revamp anyway: Changes or additions to /etc/skel don't transfer to existing accounts, so that doesn't seem to be a very good mechanism to distribute default configs like this, at least not as the sole mechanism.

Some configs (doesn't relate to these files, I think) should be determined more dynamically, too: my clock was off until I figured out that the default is to set the NEXTSPACE timezone to Kiev. 

This patch exists to make it possible to build things, because the package build step can't expect those files (absolute paths, it's not even pointing to the etc/skel/ hierarchy somewhere in the source tree! although even that is outside the Applications/ folder) to exist at that time. Once there's a more structured way on how to present initial configuration this patch can go, but it was the least intrusive fix for now.

Reply:
Transfering defaults to existing user is another story. I agree it should be enhanced. Files inside Workspace.app are WindowMaker legacy that: 1. is not changed during development 2. should be removed in the future.

/etc/skel file must be exist at the moment you build nextspace-applications package because nextspace-core is dependency for nextspace-frameworks and nextspace-applications. If Debian build system can't use files outside of build tree - well then.

Reply:
My proposal for this would be to keep this patch for debian packaging for now. I'll happily look into a more general solution for defaults, keeping in mind not to put them in Workspace.app. ok?

Reply:
No problem.